src/sys/memfd.rs

```diff
@@ -97,12 +100,13 @@ pub fn memfd_create(name: &CStr, flags: MemFdCreateFlag) -> Result<OwnedFd> {
                     target_env = "musl",
                 )))]
             {
-                libc::memfd_create(name.as_ptr(), flags.bits())
+                libc::memfd_create(cstr.as_ptr(), flags.bits())
             } else {
-                libc::syscall(libc::SYS_memfd_create, name.as_ptr(), flags.bits())
+                libc::syscall(libc::SYS_memfd_create, cstr.as_ptr(), flags.bits())
```

Comment:
Right, because the glibc wrapper may only be available since a specific version, and if you are using an older glibc and this Nix wrapper, the code won't compile. But raw syscalls are inherently unsafe, Nix tries to avoid it if possible.

Reply:
Honestly, this kind of thing has always been bothering me. I just created a tracking issue #2538 for these interfaces. May I ask about the issue you encountered with this?

Reply:
> the issue you encountered with this?

Not in real-world, but a blocker I don't use nix at https://github.com/fast/logforth/pull/80 (memfd.rs)

Reply:
Perhaps we should wrap the raw syscall. 😪

If we want to discuss this further, let's discuss it there #2538.